plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlab.ts

```diff
@@ -38,14 +38,33 @@ export function createPublishGitlabAction(options: {
   return createTemplateAction<{
     repoUrl: string;
     defaultBranch?: string;
+    /** @deprecated in favour of settings.visibility field */
     repoVisibility?: 'private' | 'internal' | 'public';
     sourcePath?: string;
     token?: string;
     gitCommitMessage?: string;
     gitAuthorName?: string;
     gitAuthorEmail?: string;
     setUserAsOwner?: boolean;
+    /** @deprecated in favour of settings.topics field */
     topics?: string[];
+    settings?: Record<string, any>;
+    branches?: Array<{
```

Comment:
What happens if I give it 5 branches, are the templated content also published to those branches or do they remain empty? I would expect it to also contain the same content by reading the description for this action.

Reply:
when you provide configuration for branches you also specify from which branch the new branch will be created using `ref` field. So all published content from `ref` branch will be in the newly created one.

We can enhance the current implementation to use `defaultBranch` value as default for newly created branches, which will make it simpler and more predictable.